src/GraphQL.Harness/Startup.cs

```diff
@@ -56,6 +56,7 @@ public void ConfigureServices(IServiceCollection services)
             // add infrastructure stuff
             services.AddHttpContextAccessor();
             services.AddLogging(builder => builder.AddConsole());
+            services.AddSingleton<GraphQLMiddleware>();
```

Comment:
How did it work before? I mean ` app.UseMiddleware<GraphQLMiddleware>();` without `services.Add`?

Reply:
All services from ctor and Invaoke are discovered and pulled by ASP.NET subsystem, right?

Reply:
Previously, yes that was correct, as it was "ASP.Net Core convention-based middleware".  It was essentially a scoped/transient service, and DI services could be added to either the constructor or Invoke method.

With this change, the middleware is pulled from DI and it is called "ASP.Net Core factory-based middleware".

I think it's just a clearer sample.

Reply:
What about middleware from server repo?

Reply:
I'll review when I look at the serialization